tests/bindings/test_measurements_nb.py

```diff
@@ -0,0 +1,403 @@
+
+    # List of modules to test
+    modules = [f"pennylane_lightning.lightning_{b}_nb" for b in backends]
+
+    @pytest.fixture
+    def get_classes(self):
+        """Get StateVector, Measurements, NamedObs, and HermitianObs classes from module based on precision."""
+
+        def _get_classes(module, precision="64"):
+            # Get StateVector class
+            state_vector_class_name = f"StateVectorC{precision}"
+            if hasattr(module, state_vector_class_name):
+                state_vector_class = getattr(module, state_vector_class_name)
+            else:
+                pytest.skip(f"Class {state_vector_class_name} not available in module")
+
+            # Get Measurements class
+            measurements_class_name = f"MeasurementsC{precision}"
+            if hasattr(module, measurements_class_name):
+                measurements_class = getattr(module, measurements_class_name)
+            else:
+                pytest.skip(f"Class {measurements_class_name} not available in module")
+
+            # Get NamedObs class
+            named_obs_class_name = f"NamedObsC{precision}"
+            if hasattr(module, named_obs_class_name):
+                named_obs_class = getattr(module, named_obs_class_name)
+            else:
+                # Try to find it in the observables submodule if it exists
+                if hasattr(module, "observables") and hasattr(
+                    module.observables, named_obs_class_name
+                ):
+                    named_obs_class = getattr(module.observables, named_obs_class_name)
+                else:
+                    pytest.skip(
+                        f"Class {named_obs_class_name} not available in module or module.observables"
+                    )
+
+            # Get HermitianObs class
+            hermitian_obs_class_name = f"HermitianObsC{precision}"
+            if hasattr(module, hermitian_obs_class_name):
+                hermitian_obs_class = getattr(module, hermitian_obs_class_name)
+            else:
+                # Try to find it in the observables submodule if it exists
+                if hasattr(module, "observables") and hasattr(
+                    module.observables, hermitian_obs_class_name
+                ):
+                    hermitian_obs_class = getattr(module.observables, hermitian_obs_class_name)
+                else:
+                    hermitian_obs_class = None
+
+            return state_vector_class, measurements_class, named_obs_class, hermitian_obs_class
+
+        return _get_classes
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_state_initialization(self, backend_name, precision, get_classes, nanobind_module):
+        """Test state vector initialization."""
+        module = nanobind_module(backend_name)
+
+        StateVectorClass, MeasurementsClass, _, _ = get_classes(module, precision)
+        # Create a numpy array representing |0> state with appropriate size
```

Comment:
We should probably test with a different state to ensure that `sv.updateData` successfully modifies the state amplitudes (assuming the constructor initializes the statevector to |00...0> anyway)